table_engine/src/partition/rule/mod.rs

```diff
@@ -48,17 +67,4 @@ pub trait PartitionRule: Send + Sync + 'static {
     fn locate_partitions_for_read(&self, filters: &[PartitionFilter]) -> Result<Vec<usize>>;
 }
 
-#[allow(dead_code)]
-#[derive(Debug)]
-pub struct ColumnWithType {
-    column: String,
-    datum_type: DatumKind,
-}
-
-impl ColumnWithType {
-    pub fn new(column: String, datum_type: DatumKind) -> Self {
-        Self { column, datum_type }
-    }
-}
-
-pub type PartitionRuleRef = Box<dyn PartitionRule>;
+pub type PartitionRulePtr = Box<dyn PartitionRule>;
```

Comment:
IMO `PartitionRuleRef` is better, in safe Rust we don't operate on pointers.

Reply:
I guess `Ref` is usually used for `Arc`?